File: src/main/java/io/sommers/aiintheipaw/model/sprint/Sprint.java
package io.sommers.aiintheipaw.model.sprint;

import io.sommers.aiintheipaw.model.channel.IChannel;

public class Sprint {
    private final long id;
    private final IChannel channel;

    private SprintStatus status;

    public Sprint(long id, IChannel channel, SprintStatus status) {
        this.id = id;
        this.channel = channel;
        this.status = status;
    }

    public long getId() {
        return id;
    }

    public IChannel getChannel() {
        return channel;
    }

    public SprintStatus getStatus() {
        return status;
    }

    public void setStatus(SprintStatus status) {
        this.status = status;
    }
}

File: src/main/scala/model/sprint/Sprint.scala
package io.sommers.aiintheipaw
package model.sprint

import model.channel.Channel
import model.user.User

import zio.Duration

import java.time.Instant

case class Sprint(
  id: Long,
  channel: Channel,
  startedBy: User,
  sections: Seq[SprintSection],
  entries: Seq[SprintEntry]
)

case class SprintSection(
  id: Long,
  status: SprintStatus,
  startTime: Instant,
  duration: Duration
)

case class SprintEntry(
  id: Long,
  user: User,
  startSectionId: Long,
  startingWords: Long,
  endingWords: Option[Long]
)